### DISCUSSION TITLE
Passing ActivityOutput to correlation activity

### DISCUSSION THREAD

**Discussion by kAlexPodles** [elsa-workflows/elsa-core#3400] [Q&A]
Hello Elsa Community!

 I am trying to build a workflow which should pass some activity output to correlation, but i have a problem with reading the correlation value.
There is a similar activity to this one https://github.com/elsa-workflows/elsa-core/blob/master/src/activities/Elsa.Activities.File/Activities/ReadFile/ReadFile.cs (but the outcome is a string).

AFAIU, it should be possible to read a value from prev activity like this:
![image](https://user-images.githubusercontent.com/103921304/199314591-ff79a6e1-a2d0-4813-919b-89bdfdfe1eef.png)

But for some reason, the correlation field is always empty;
![image](https://user-images.githubusercontent.com/103921304/199315306-9aab6e04-1343-4027-a2be-8ae69ee271b8.png)

**Comment by FransVanEk** [comment]
In Javascript you can indeed access the output of a previous activity

[javascript expressions](https://elsa-workflows.github.io/elsa-core/docs/expressions/expressions-javascript)

some pointers:

- You can use activities.<Name of the activity>.Output()
- don't forget to name the activity. In programmatic mode .WithName("<Name of the activity>")
